src/Psalm/Internal/Codebase/Populator.php

```diff
@@ -181,10 +181,17 @@ private function populateClassLikeStorage(ClassLikeStorage $storage, array $depe
 
         $dependent_classlikes[$fq_classlike_name_lc] = true;
 
-        $this->populateDataFromTraits($storage, $storage_provider, $dependent_classlikes);
+        foreach ($storage->used_traits as $used_trait_lc => $_) {
+            $this->populateDataFromTrait($storage, $storage_provider, $dependent_classlikes, $used_trait_lc);
+        }
 
         if ($storage->parent_classes) {
-            $this->populateDataFromParentClass($storage, $storage_provider, $dependent_classlikes);
+            $this->populateDataFromParentClass(
+                $storage,
+                $storage_provider,
+                $dependent_classlikes,
+                reset($storage->parent_classes)
```

Comment:
was there a reason to not loop over parent classes here?

Reply:
nevermind, that was the previous behavior, as I saw below

Reply:
That corresponds with https://github.com/vimeo/psalm/pull/7110/files/8ceb0e2077a07d01932ae350b8edac58b11eb9bf#diff-76daa06f11322c5f544716df14e1c64a90a9ee87e749369914f2a3dbbf2e5f5bL483, it was moved directly without changing. I assume it's fine because `$storage->parent_classes` is supposed to be ordered with the directly extended class first, and that class's storage should already have been populated with its parent data.

Reply:
That corresponds with https://github.com/vimeo/psalm/pull/7110/files/8ceb0e2077a07d01932ae350b8edac58b11eb9bf#diff-76daa06f11322c5f544716df14e1c64a90a9ee87e749369914f2a3dbbf2e5f5bL483.
It only needs to populate the direct parent, because the direct parent will already have been populated by its parent.